common/djangoapps/util/testing.py

```diff
@@ -28,16 +28,17 @@ class UrlResetMixin(object):
     Doing this is expensive, so it should only be added to tests that modify settings
     that affect the contents of urls.py
     """
-
+    URLS_AUTO_RESET = True
```

Comment:
1: I agree with the name change, will do.

2 & 3: The reason I didn't do that was because the various setups and teardowns want to have certain settings overrides applied, which they apply differently (some `patch` it, some use `override_settings` at the method or even class level). Some test cases explicitly do not call their super-class's `setUpClass` method because they invoke `setUpClassAndTestData`, which will call `ModuleStoreTestCase` and `SharedModuleStoreTestCase` `setUpClass`, but not `setUpClass` on other mixins that are siblings in the hierarchy. So instead of trying to sort all that out, this seemed simpler.

On that note, I think that we have a lot of inherited magic in our tests, and I'd like to see us to push more on the side of explicit and locally obvious, even if it's a bit more verbose. I think that it's fine that `ModuleStoreTestCase` and `SharedModuleStoreTestCase` do all that init work in their setUps, but I'd like to see supplementary mixins offer functionality that needs to be explicitly invoked by the leaf node classes that get executed, rather than things that automatically fire. Remember how there was that item in Effective Python suggesting that multiple inheritance should usually be confined to mixin classes that don't need to have `__init__()` run? I'm beginning to think of test `setUpClass` in much the same way.

This is PR is a bit of a mishmash of thoughts because I was thinking about these things as I was working through it. I'll try to fix it up a bit and resubmit for review later this week.

Thank you.